src/client/NuGet.CommandLine.StagingExtensions/StageCommand.cs

```diff
@@ -0,0 +1,135 @@
+        [Option(typeof(StagingResources), "StageCommandCreateDescription")]
+        public string Create { get; set; }
+
+        [Option(typeof(StagingResources), "StageCommandDropDescription")]
+        public string Drop { get; set; }
+
+        [Option(typeof(StagingResources), "StageCommandSourceDescription")]
+        public string Source { get; set; }
+
+        [Option(typeof(StagingResources), "StageCommandApiKeyDescription")]
+        public string ApiKey { get; set; }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            // Verify input
+            if (!string.IsNullOrEmpty(Create) && !string.IsNullOrEmpty(Drop) ||
+                (string.IsNullOrEmpty(Create) && string.IsNullOrEmpty(Drop)))
+            {
+                HelpCommand.ViewHelpForCommand(CommandAttribute.CommandName);
+                return;
+            }
+
+            // Get source - from command line param or default from config file
+            string source = ResolveSource();
+
+            // Get api key - from command line param or from config file
+            string apiKey = GetApiKey(source);
+
+            var stageManagementResource = await GetStageManagementResource(source);
+
+            if (!string.IsNullOrEmpty(Create))
+            {
+                await CreateStage(stageManagementResource, apiKey, Create);
+            }
+        }
+
+        private async Task CreateStage(StageManagementResource stageManagementResource, string apiKey, string stageDisplayName)
+        {
+            Console.LogInformation(string.Format(CultureInfo.CurrentCulture, StagingResources.CreatingStageMessage, stageDisplayName));
+
+            var createResult = await stageManagementResource.Create(stageDisplayName, apiKey, Console);
+
+            Console.LogInformation(string.Format(CultureInfo.CurrentCulture, StagingResources.CreatedStageMessage, createResult.Feed));
+        }
+
+        private async Task<StageManagementResource> GetStageManagementResource(string source)
+        {
+            var sourceRepository = GetSourceRepository(new Configuration.PackageSource(source));
+            var stageManagementResourceProvider = await sourceRepository.GetResourceAsync<StageManagementResource>();
+            return stageManagementResourceProvider;
+        }
+
+        private SourceRepository GetSourceRepository(Configuration.PackageSource source)
+        {
+            var resourceProviders = new List<Lazy<INuGetResourceProvider>>();
+            resourceProviders.AddRange(FactoryExtensionsV2.GetCoreV3(Repository.Provider));
+            resourceProviders.Add(new Lazy<INuGetResourceProvider>(() => new StageManagementResourceProvider()));
+
+            return new SourceRepository(source, resourceProviders);
+        }
+
+        private string ResolveSource()
+        {
+            string source = Source;
+
+            if (string.IsNullOrEmpty(source))
+            {
+                source = SourceProvider.DefaultPushSource;

```

Comment:
In 3.4.0 nuget.exe removed defaulting to nuget.org as the push source. You probably want to avoid it for staging also.